html/janus.js

```diff
@@ -988,7 +990,20 @@ function Janus(gatewayCallbacks) {
 				ws.close();
 				ws = null;
 			} else {
-				navigator.sendBeacon(server + "/" + sessionId, JSON.stringify(request));
+        var uri = server + "/" + sessionId;
+        if (unloadParams) {
+          Object.keys(unloadParams)
+            .filter(function(key) { return unloadParams[key] != null; }) // Ignore null or undefined params
+            .forEach(function(key) {
+              // If there are more than 1 param, we need to use '&' instead of '?' after first param
+              if (uri.search(/\?.*=.*/) >= 0) {
+                uri += '&' + key + '=' + unloadParams[key];
+              } else {
+                uri += '?' + key + '=' + unloadParams[key];
+              }
+            });
+        }
+				navigator.sendBeacon(uri, JSON.stringify(request));
```

Comment:
Not sure what's the point of adding those query string arguments, if we're still sending the request to Janus. The Janus integrated web server will ignore them all.

Reply:
This PR was made to use params in URI for `sendBeacon` if the Janus server is not achieved directly, for example using something like a proxy. In my own case, we have developed an admin panel to handle all web requests, so janus is never reached directly (to prevent unwanted access/attacks to the public server and only access through our authentication system). That made us to improve how calls are made in JS client to reach janus and we found that the `sendBeacon` URI was not "customizable".

Indeed, as discussed in the related issue, we could create our own `sendBeacon`, but we thought that someone who wanted to use janus behind any kind of proxy or redirecting requests could find an easy way to implement the current janus.js unload function in order to prevent doing themselves (e.g. they don't know how to do it or w/e).

As a side note, I'll like to mention that this was just to know if it could be useful to be implemented. Just let me know if you think this couldn't be useful at all in a day-by-day scenario.

Reply:
I've given this some thought and I don't think this is generic enough to be part of janus.js, sorry. This seems more specific to your use case, which is fine, but doesn't mean it should be part of the repo. Closing.